ivy/functional/backends/paddle/device.py

```diff
@@ -17,62 +19,104 @@
 def dev(x: paddle.Tensor, /, *, as_native: bool = False) -> Union[ivy.Device, Place]:
     dv = x.place
     if as_native:
-        if isinstance(dv, Place):
-            dv = "gpu" if dv.is_gpu_place() else "cpu"
-        return x.place
+        return dv
     return as_ivy_dev(dv)
 
 
 def to_device(
     x: paddle.Tensor,
-    device: Place,
+    device: Union[ivy.Device, Place],
     /,
     *,
     stream: Optional[int] = None,
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
-    if device is None:
-        return x
-    # TODO: check memory leak because ret is copying the original tensor
-    ret = paddle.to_tensor(
-        x, place=as_native_dev(device), stop_gradient=x.stop_gradient
-    )
-    return ret
+    if device is not None:
+        native_dev = as_native_dev(device)
+        if not dev(x, as_native=True)._equals(native_dev):  # type: ignore
+            # TODO: check memory leak because ret is copying the original tensor
+            return paddle.to_tensor(x, place=native_dev, stop_gradient=x.stop_gradient)
+    return x
 
 
-def as_ivy_dev(device: Place, /):
+def as_ivy_dev(device, /):
```

Comment:
ig the typehint was removed by mistake?

Reply:
No, not really. Paddle Place is not the base type for all devices. There are more device types like CPUPlace and CUDAPlace which are native devices. So Place isn't the class it supports.

Reply:
Oh okay, ig then we should be doing a `Union[]` of relevant place types as the type-hint here 🙂